src/sentry/grouping/component.py

```diff
@@ -28,25 +29,22 @@ def _calculate_contributes[ValuesType](values: Sequence[ValuesType]) -> bool:
     return False
 
 
-class BaseGroupingComponent[ValuesType: str | int | BaseGroupingComponent[Any]]:
+class BaseGroupingComponent[ValuesType: str | int | BaseGroupingComponent[Any]](ABC):
     """A grouping component is a recursive structure that is flattened
     into components to make a hash for grouping purposes.
     """
 
-    id: str = "default"
     hint: str | None = None
     contributes: bool = False
     values: Sequence[ValuesType]
```

Comment:
`id: str = "some literal"` is better written as just `id = "some literal"`
.  
the rest of these should not appear on the class itself but in `__init__` -- putting them here means you have to specify them at least twice (sometimes three times) and they'll drift.  best case you have to specify them exactly once (in parameters)